src/Daemon/Daemon.cpp

```diff
@@ -119,16 +119,53 @@ namespace usbguard
     _config.close();
   }
 
-  void Daemon::loadConfiguration(const std::string& path)
+  int Daemon::checkPermissions(const std::string& path,
+                               const mode_t permissions)
+  {
+    struct stat file_stat;
+    mode_t permission_bad { ( S_IRWXU | S_IRWXG | S_IRWXO ) - permissions };
+
+    if(!stat(path.c_str(), &file_stat)) {
+      if (S_ISREG(file_stat.st_mode)) {
+        if (  !( file_stat.st_mode & permissions ) ||
+               ( file_stat.st_mode & permission_bad)
+           )
+        {
+          USBGUARD_LOG(Error) << "Policy leaks are possible with this permissions!";
+          throw Exception("Check permissions", path, "Policy leaks possible");
+        }
+        else {
+          USBGUARD_LOG(Info) << "File has correct permissions.";
+        }
+      }
+      else {
+        USBGUARD_LOG(Error) << "ERROR: In specified path is not located file.";
+        throw Exception("Check permissions", path, "File not exists");
+      }
+    }
+    else {
+      USBGUARD_LOG(Error) << "ERROR: obtaining file permissions!";
+      throw Exception("Check permissions", path, "Get file permissions");
```

Comment:
It would be better to use the ErrnoException here (see https://github.com/dkopecek/usbguard/blob/master/src/Library/public/usbguard/Exception.hpp#L112):
```
throw ErrnoException("Check permissions", path, errno);
```

Also include the errno value in the log output.